Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/DebugLogger.kt

```diff
@@ -165,26 +165,37 @@ class DebugLogger(
                     delay(1000)
 
                     val formattedMessage = rawLine.format()
-                    val censored: Collection<BugCensor.CensoredString> = bugCensors.get()
+                    val censored: Collection<BugCensor.CensorContainer> = bugCensors.get()
                         .map {
                             async {
-                                it.checkLog(formattedMessage)
+                                try {
+                                    it.checkLog(formattedMessage)
+                                } catch (e: Exception) {
+                                    Log.e(TAG, "Error in censor module $it", e)
+                                    null
+                                }
                             }
                         }
                         .awaitAll()
                         .filterNotNull()
-                        .filter { it.range != null }
 
                     val toWrite: String = when (censored.size) {
                         0 -> formattedMessage
-                        1 -> censored.single().string
-                        else -> {
-                            val minMin = censored.minOf { it.range!!.first }
-                            val maxMax = censored.maxOf { it.range!!.last }
-                            formattedMessage.replaceRange(minMin, maxMax, CENSOR_COLLISION_PLACERHOLDER)
-                        }
+                        1 -> censored.single().compile()?.censored ?: formattedMessage
```

Comment:
`compile` can never return null here, so maybe use `!!`?

Reply:
Only if all censors return non-empty containers though, the "non-null" is not enforced by the `DebugLogger` but by other classes, to prevent crashes due to developer mistakes, I'd prefer to keep it null-safe here.